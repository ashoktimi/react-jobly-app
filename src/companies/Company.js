import { useState, useEffect } from 'react';
import { useParams, useNavigate   } from "react-router-dom";
import JoblyApi from '../api/api'
import JobCardList from "../jobs/JobCardList";
import LoadingSpinner from "../helpers/LoadingSpinner";

const Company = () =>{
    const { id } = useParams();
    const navigate = useNavigate()
    const [company, setCompany] = useState ({})
 
    useEffect(() => {
      async function getCompany (handle){
        try {
            let data = await JoblyApi.getCompany(handle);
            setCompany(data);
          } catch (error) {
            console.log(error);
            alert(`${id} not found.`)
            navigate('/companies')
          }
      }
      getCompany(id)
    }, [id, navigate]);

    if (!company) return <LoadingSpinner />;

    return(
    <div>
        <p>{company.name}</p>
        <p>{company.description}</p>        
        <div>
            {company.jobs && company.jobs.length > 0 
            ? (<JobCardList jobs={company.jobs} />)
            :(
            <div>
              <p>{id} not found.</p>
            </div>
          )}
        </div>
    </div>
    )
}

export default Company;
