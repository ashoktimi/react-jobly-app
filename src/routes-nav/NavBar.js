
import React, {useContext} from "react";
import { NavLink } from "react-router-dom";
import { Navbar, Nav } from 'react-bootstrap';
import UserContext from "../auth/userContext";
import "./Navigation.css";

function NavBar({ logout }) {
  const { currentUser } = useContext(UserContext);
  console.debug("Navigation", "currentUser=", currentUser);

  function loggedInNav() {
    return(
      <ul>
          <Nav.Item>
            <NavLink to="/companies">Companies</NavLink>
          </Nav.Item>
          <Nav.Item>
            <NavLink to="/jobs">Jobs</NavLink>
          </Nav.Item>
          <Nav.Item>
            <NavLink to="/profile"> Profile </NavLink>
          </Nav.Item>
          <Nav.Item>
            <NavLink to="/" onClick={logout}> Log out {currentUser.first_name || currentUser.username} </NavLink>
          </Nav.Item>
      </ul>
    )
  }

  function loggedOutNav(){
    return(
      <ul>
          <Nav.Item>
            <NavLink to="/login"> Login </NavLink>
          </Nav.Item>
          <Nav.Item>
            <NavLink to="/signup"> Signup </NavLink>
          </Nav.Item>
      </ul>
    )
  }

  return (
    <div>
      <Navbar expand="md">
        <NavLink to='/'> Jobly </NavLink>
        {currentUser ? loggedInNav() : loggedOutNav()}
      </Navbar>
    </div>
  );
}

export default NavBar;