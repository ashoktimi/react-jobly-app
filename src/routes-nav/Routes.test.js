import React from "react";
import { render } from "@testing-library/react";
import Routers from "./Routers";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";
import NavBar from './NavBar';
import Company from '../companies/Company';
import Login from '../auth/LoginForm';
import CompanyList from '../companies/CompanyList';

it("renders without crashing", function () {
  render(
      <MemoryRouter>
          <Routers />
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
          <Routers />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});



test('NavBar component renders correctly', () => {
  const { container } = render(
  <MemoryRouter>
    <UserProvider>
      <NavBar />
    </UserProvider>   
  </MemoryRouter>  
  );
  expect(container).toMatchSnapshot();
});

test('Company component renders correctly', () => {
  const { container } = render(
   <MemoryRouter>
    <UserProvider>
     <Company />
    </UserProvider>   
  </MemoryRouter> 
  );
  expect(container).toMatchSnapshot();
});


test('Routes to login page', () => {
  const { getByText } = render(
    <MemoryRouter>
     <UserProvider>
      <Login />
    </UserProvider>  
    </MemoryRouter>
  );
  expect(getByText("Log In")).toBeInTheDocument();
});