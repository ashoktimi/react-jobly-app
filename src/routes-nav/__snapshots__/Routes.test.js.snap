// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company component renders correctly 1`] = `
<div>
  <div>
    <p />
    <p />
    <div>
      <div>
        <p>
           not found.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBar component renders correctly 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-md navbar-light"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
         Jobly 
      </a>
      <ul>
        <div
          class="nav-item"
        >
          <a
            class=""
            href="/companies"
          >
            Companies
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class=""
            href="/jobs"
          >
            Jobs
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class=""
            href="/profile"
          >
             Profile 
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
             Log out 
            testfirst
             
          </a>
        </div>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Routers component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <nav
        class="navbar navbar-expand-md navbar-light"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
           Jobly 
        </a>
        <ul>
          <div
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
               Login 
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class=""
              href="/signup"
            >
               Signup 
            </a>
          </div>
        </ul>
      </nav>
    </div>
    <div
      class="Homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-4 font-weight-bold"
        >
          Jobly
        </h1>
        <p
          class="lead"
        >
          All the jobs in one, convenient place.
        </p>
        <p>
          <a
            class="btn btn-primary font-weight-bold mr-3"
            href="/login"
          >
            Log in
          </a>
          <a
            class="btn btn-primary font-weight-bold"
            href="/signup"
          >
            Sign up
          </a>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <nav
        class="navbar navbar-expand-md navbar-light"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
           Jobly 
        </a>
        <ul>
          <div
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
               Login 
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class=""
              href="/signup"
            >
               Signup 
            </a>
          </div>
        </ul>
      </nav>
    </div>
    <div
      class="Homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-4 font-weight-bold"
        >
          Jobly
        </h1>
        <p
          class="lead"
        >
          All the jobs in one, convenient place.
        </p>
        <p>
          <a
            class="btn btn-primary font-weight-bold mr-3"
            href="/login"
          >
            Log in
          </a>
          <a
            class="btn btn-primary font-weight-bold"
            href="/signup"
          >
            Sign up
          </a>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
